src/main/webapp/js/instructorFeedbackEdit/visibilityOptions.js

```diff
@@ -142,41 +152,68 @@ function disableRow($containingForm, row) {
     $tdElements.parent().hide();
 }
 
-function updateEditTabAccordingToRecipient($containingForm) {
+function disableRowsAccordingToRecipient($containingForm) {
     var recipientType = $containingForm.find('select[name="recipienttype"]').val();
-    if (isRecipientsTeamMembersVisibilityOptionInvalidForRecipientType(recipientType)) {
-        // show the row Recipient(s) and hide the row Recipient's Team Members
-        enableRow($containingForm, ROW_RECIPIENT);
+    switch (recipientType) {
+    case 'SELF':
+        // ROW_RECIPIENT is disabled because self-feedback is always visible to giver
+        disableRow($containingForm, ROW_RECIPIENT);
+        // ROW_RECIPIENT_TEAM is disabled because it is the same as ROW_GIVER_TEAM
         disableRow($containingForm, ROW_RECIPIENT_TEAM);
-        return;
-    } else if (recipientType === 'NONE') {
-        // hide both the row Recipient(s) and the row Recipient's Team Members
+        break;
+    case 'STUDENTS':
+        // all options enabled when recipientType is STUDENTS (subject to options disabled by giverType)
+        break;
+    case 'OWN_TEAM':
+        // ROW_RECIPIENT is disabled because it is the same as ROW_GIVER_TEAM
+        disableRow($containingForm, ROW_RECIPIENT);
+        // ROW_RECIPIENT_TEAM is disabled because it is same as ROW_RECIPIENT
+        disableRow($containingForm, ROW_RECIPIENT_TEAM);
+        break;
+    case 'INSTRUCTORS':
+    case 'TEAMS':
+    case 'OWN_TEAM_MEMBERS':
+    case 'OWN_TEAM_MEMBERS_INCLUDING_SELF':
+        // ROW_RECIPIENT_TEAM is disabled because it is same as ROW_RECIPIENT
         disableRow($containingForm, ROW_RECIPIENT_TEAM);
+        break;
+    case 'NONE':
         disableRow($containingForm, ROW_RECIPIENT);
-        return;
+        disableRow($containingForm, ROW_RECIPIENT_TEAM);
+        break;
+    default:
+        throw 'Unexpected receiverType';
     }
-    
-    enableRow($containingForm, ROW_RECIPIENT);
-    enableRow($containingForm, ROW_RECIPIENT_TEAM);
 }
 
-/**
- * Returns true if "recipient's team members" visibility option
- * is not applicable for the recipient type
- */
-function isRecipientsTeamMembersVisibilityOptionInvalidForRecipientType(recipientType) {
-    return recipientType === 'OWN_TEAM' || recipientType === 'TEAMS'
-           || recipientType === 'INSTRUCTORS' || recipientType === 'OWN_TEAM_MEMBERS'
-           || recipientType === 'OWN_TEAM_MEMBERS_INCLUDING_SELF';
+function disableRowsAccordingToGiver($containingForm) {
+    var giverType = $containingForm.find('select[name="givertype"]').val();
+    switch (giverType) {
+    case 'STUDENTS':
+        // all options enabled when giverType is STUDENTS (subject to options disabled by recipientType)
+        break;
+    case 'SELF':
+    case 'INSTRUCTORS':
+    case 'TEAMS':
+        // ROW_GIVER_TEAM is disbled because it is the same as ROW_INSTRUCTORS
```

Comment:
`disbled` should be `disabled`

Also, this comment doesn't apply to `case 'TEAMS'` too right? Maybe we could do something like
```
case 'SELF':
case 'INSTRUCTORS':
    // ROW_GIVER_TEAM is disabled for SELF and INSTRUCTORS because it is the same as ROW_INSTRUCTORS
case 'TEAMS':
    // ROW_GIVER_TEAM is disabled for TEAMS as it is an invalid User/Group choice for the giverType
    
    disableRow($containingForm, ROW_GIVER_TEAM);
    break;
```